From patchwork Thu Aug  4 09:15:16 2016
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [3/4] KVM: arm/arm64: Check for broadcast TLBI support
From: Matthias Brugger <mbrugger@suse.com>
X-Patchwork-Id: 9263097
Message-Id: <1470302117-32296-4-git-send-email-mbrugger@suse.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com, christoffer.dall@linaro.org,
 marc.zyngier@arm.com, linux@armlinux.org.uk, catalin.marinas@arm.com, 
 will.deacon@arm.com
Cc: mark.rutland@arm.com, mbrugger@suse.com, kvm@vger.kernel.org,
 david.daney@cavium.com, ard.biesheuvel@linaro.org, zlim.lnx@gmail.com,
 suzuki.poulose@arm.com, agraf@suse.de, linux-kernel@vger.kernel.org, 
 rrichter@cavium.com, lorenzo.pieralisi@arm.com, james.morse@arm.com, 
 dave.long@linaro.org, kvmarm@lists.cs.columbia.edu,
 linux-arm-kernel@lists.infradead.org
Date: Thu,  4 Aug 2016 11:15:16 +0200
Patch-mainline: Submitted, https://patchwork.kernel.org/patch/9263097
References: bsc#986099

KVM awaits a working broadcast TLBI implementation.
Error out early when this is not provided by the
hardware.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 virt/kvm/arm/arm.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/virt/kvm/arm/arm.c
+++ b/virt/kvm/arm/arm.c
@@ -1429,6 +1429,13 @@ int kvm_arch_init(void *opaque)
 	int err;
 	int ret, cpu;
 
+#ifdef CONFIG_ARM64
+	if (cpus_have_cap(ARM64_HAS_NO_BCAST_TLBI)) {
+		kvm_err("Missing broadcast TLBI support.\n");
+		return -ENODEV;
+	}
+#endif
+
 	if (!is_hyp_mode_available()) {
 		kvm_err("HYP mode not available\n");
 		return -ENODEV;
